diff --git a/v8/src/inspector/v8-runtime-agent-impl.cc b/v8/src/inspector/v8-runtime-agent-impl.cc
--- a/v8/src/inspector/v8-runtime-agent-impl.cc
+++ b/v8/src/inspector/v8-runtime-agent-impl.cc

@@ -1055,28 +1055,7 @@
 }
 
 void V8RuntimeAgentImpl::addBindings(InspectedContext* context) {
-  const String16 contextName = context->humanReadableName();
-  if (!m_enabled) return;
-
-  protocol::DictionaryValue* globalBindings =
-      m_state->getObject(V8RuntimeAgentImplState::globalBindings);
-  if (globalBindings) {
-    for (size_t i = 0; i < globalBindings->size(); ++i) {
-      addBinding(context, globalBindings->at(i).first);
-    }
-  }
-
-  protocol::DictionaryValue* contextBindings =
-      m_state->getObject(V8RuntimeAgentImplState::contextBindings);
-  if (contextBindings) {
-    protocol::DictionaryValue* bindings =
-        contextBindings->getObject(contextName);
-    if (bindings) {
-      for (size_t i = 0; i < bindings->size(); ++i) {
-        addBinding(context, bindings->at(i).first);
-      }
-    }
-  }
+  return;
 }
 
 void V8RuntimeAgentImpl::restore() {
@@ -1198,7 +1177,6 @@
 }
 
 void V8RuntimeAgentImpl::messageAdded(V8ConsoleMessage* message) {
-  if (m_enabled) reportMessage(message, true);
 }
 
 bool V8RuntimeAgentImpl::reportMessage(V8ConsoleMessage* message,


diff --git a/v8/src/inspector/v8-runtime-agent-impl.h b/v8/src/inspector/v8-runtime-agent-impl.h
--- a/v8/src/inspector/v8-runtime-agent-impl.h
+++ b/v8/src/inspector/v8-runtime-agent-impl.h

@@ -160,7 +160,7 @@
                std::unique_ptr<protocol::DictionaryValue> hints,
                int executionContextId);
   void messageAdded(V8ConsoleMessage*);
-  bool enabled() const { return m_enabled; }
+  bool enabled() const { return false; }
 
  private:
   bool reportMessage(V8ConsoleMessage*, bool generatePreview);

