diff --git a/headless/lib/browser/headless_browser_impl.cc b/headless/lib/browser/headless_browser_impl.cc
--- a/headless/lib/browser/headless_browser_impl.cc
+++ b/headless/lib/browser/headless_browser_impl.cc

@@ -70,7 +70,7 @@
 namespace {
 
 // Product name for building the default user agent string.
-const char kHeadlessProductName[] = "HeadlessChrome";
+const char kHeadlessProductName[] = "Chrome";
 
 #if defined(HEADLESS_USE_PREFS)
 const base::FilePath::CharType kLocalStateFilename[] =

