--- /tmp/chromium_orig/third_party/blink/renderer/core/frame/local_dom_window.cc	2026-05-27 04:39:02
+++ /Volumes/Lexar-1T/mk_ch/chromium/src/third_party/blink/renderer/core/frame/local_dom_window.cc	2026-08-01 14:48:12
@@ -32,11 +32,13 @@
 
 #include "base/command_line.h"
 #include "base/metrics/histogram_macros.h"
+#include "base/strings/string_number_conversions.h"
 #include "base/task/single_thread_task_runner.h"
 #include "base/trace_event/trace_id_helper.h"
 #include "base/trace_event/typed_macros.h"
 #include "build/build_config.h"
 #include "cc/input/snap_selection_strategy.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 #include "net/storage_access_api/status.h"
 #include "third_party/blink/public/common/features.h"
@@ -1574,6 +1576,21 @@
   if (!GetFrame())
     return 0;
 
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint)) {
+    int width = 1920, height = 1080;
+    if (command_line->HasSwitch(switches::kFingerprintScreenWidth))
+      width = std::stoi(
+          command_line->GetSwitchValueASCII(
+              switches::kFingerprintScreenWidth));
+    if (command_line->HasSwitch(switches::kFingerprintScreenHeight))
+      height = std::stoi(
+          command_line->GetSwitchValueASCII(
+              switches::kFingerprintScreenHeight));
+    return height;
+  }
+
   LocalFrame* frame = GetFrame();
 
   // FencedFrames should return innerHeight to prevent passing
@@ -1598,6 +1615,21 @@
 int LocalDOMWindow::outerWidth() const {
   if (!GetFrame())
     return 0;
+
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint)) {
+    int width = 1920, height = 1080;
+    if (command_line->HasSwitch(switches::kFingerprintScreenWidth))
+      width = std::stoi(
+          command_line->GetSwitchValueASCII(
+              switches::kFingerprintScreenWidth));
+    if (command_line->HasSwitch(switches::kFingerprintScreenHeight))
+      height = std::stoi(
+          command_line->GetSwitchValueASCII(
+              switches::kFingerprintScreenHeight));
+    return width;
+  }
 
   LocalFrame* frame = GetFrame();
 
@@ -1668,6 +1700,11 @@
 }
 
 int LocalDOMWindow::screenX() const {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint))
+    return 0;
+
   LocalFrame* frame = GetFrame();
   if (!frame)
     return 0;
@@ -1686,6 +1723,11 @@
 }
 
 int LocalDOMWindow::screenY() const {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint))
+    return 0;
+
   LocalFrame* frame = GetFrame();
   if (!frame)
     return 0;
--- /tmp/chromium_orig/third_party/blink/renderer/core/css/media_values.cc	2026-05-27 04:39:02
+++ /Volumes/Lexar-1T/mk_ch/chromium/src/third_party/blink/renderer/core/css/media_values.cc	2026-08-01 14:48:32
@@ -4,6 +4,9 @@
 
 #include "third_party/blink/renderer/core/css/media_values.h"
 
+#include "base/command_line.h"
+#include "base/strings/string_number_conversions.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "third_party/blink/public/common/css/scripting.h"
 #include "third_party/blink/renderer/core/css/css_resolution_units.h"
 #include "third_party/blink/renderer/core/css/css_to_length_conversion_data.h"
@@ -193,6 +196,13 @@
 }
 
 float MediaValues::CalculateDevicePixelRatio(LocalFrame* frame) {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprintDeviceScaleFactor)) {
+    return std::stof(
+        command_line->GetSwitchValueASCII(
+            switches::kFingerprintDeviceScaleFactor));
+  }
   return frame->DevicePixelRatio();
 }
 
@@ -353,6 +363,10 @@
     LocalFrame* frame) {
   DCHECK(frame);
   DCHECK(frame->GetSettings());
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint))
+    return mojom::blink::PointerType::kFine;
   return frame->GetSettings()->GetPrimaryPointerType();
 }
 
@@ -366,6 +380,10 @@
     LocalFrame* frame) {
   DCHECK(frame);
   DCHECK(frame->GetSettings());
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint))
+    return mojom::blink::HoverType::kHover;
   return frame->GetSettings()->GetPrimaryHoverType();
 }
 
@@ -385,6 +403,10 @@
 ColorSpaceGamut MediaValues::CalculateColorGamut(LocalFrame* frame) {
   DCHECK(frame);
   DCHECK(frame->GetPage());
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(switches::kFingerprint))
+    return ColorSpaceGamut::kSRGB;
   const MediaFeatureOverrides* overrides =
       frame->GetPage()->GetMediaFeatureOverrides();
   std::optional<ColorSpaceGamut> override_value =
