diff --git a/third_party/blink/renderer/core/frame/navigator_language.cc b/third_party/blink/renderer/core/frame/navigator_language.cc
--- a/third_party/blink/renderer/core/frame/navigator_language.cc
+++ b/third_party/blink/renderer/core/frame/navigator_language.cc
@@ -5,6 +5,8 @@
 #include "third_party/blink/renderer/core/frame/navigator_language.h"
 
 #include "base/command_line.h"
+#include "base/strings/string_util.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "services/network/public/cpp/features.h"
 #include "third_party/blink/public/common/switches.h"
 #include "third_party/blink/renderer/core/frame/web_feature.h"
@@ -37,10 +39,21 @@
     : execution_context_(execution_context) {}
 
 AtomicString NavigatorLanguage::language() {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint)) {
+    return AtomicString("en-US");
+  }
   return AtomicString(languages().front());
 }
 
 const Vector<String>& NavigatorLanguage::languages() {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint)) {
+    DEFINE_STATIC_LOCAL(Vector<String>, kLanguages, ({"en-US", "en"}));
+    return kLanguages;
+  }
   EnsureUpdatedLanguage();
   return languages_;
 }
