--- /tmp/chromium_orig/third_party/blink/renderer/core/events/navigator_events.cc	2026-05-27 04:39:02
+++ /Volumes/Lexar-1T/mk_ch/chromium/src/third_party/blink/renderer/core/events/navigator_events.cc	2026-08-01 16:24:41
@@ -30,6 +30,8 @@
 
 #include "third_party/blink/renderer/core/events/navigator_events.h"
 
+#include "base/command_line.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
 #include "third_party/blink/renderer/core/frame/local_frame.h"
 #include "third_party/blink/renderer/core/frame/navigator.h"
@@ -38,6 +40,10 @@
 namespace blink {
 
 int32_t NavigatorEvents::maxTouchPoints(Navigator& navigator) {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint))
+    return 0;
   LocalDOMWindow* window = navigator.DomWindow();
   return window ? window->GetFrame()->GetSettings()->GetMaxTouchPoints() : 0;
 }
