--- /tmp/chromium_orig/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc	2026-05-27 04:39:02
+++ /Volumes/Lexar-1T/mk_ch/chromium/src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc	2026-08-01 16:24:45
@@ -20,6 +20,8 @@
 
 #include "third_party/blink/renderer/modules/plugins/dom_plugin_array.h"
 
+#include "base/command_line.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "third_party/blink/public/common/features.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
 #include "third_party/blink/renderer/core/frame/local_frame.h"
@@ -73,10 +75,18 @@
 }
 
 unsigned DOMPluginArray::length() const {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint))
+    return 0;
   return dom_plugins_.size();
 }
 
 DOMPlugin* DOMPluginArray::item(unsigned index) {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint))
+    return nullptr;
   if (index >= dom_plugins_.size()) {
     return nullptr;
   }
@@ -84,6 +94,10 @@
 }
 
 DOMPlugin* DOMPluginArray::namedItem(const AtomicString& property_name) {
+  const base::CommandLine* command_line =
+      base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch(::switches::kFingerprint))
+    return nullptr;
   for (const auto& plugin : dom_plugins_) {
     if (plugin->name() == property_name) {
       return plugin.Get();
