diff --git a/third_party/blink/renderer/modules/speech/speech_synthesis.cc b/third_party/blink/renderer/modules/speech/speech_synthesis.cc
index 2327015..bd6f518 100644
--- a/third_party/blink/renderer/modules/speech/speech_synthesis.cc
+++ b/third_party/blink/renderer/modules/speech/speech_synthesis.cc
@@ -27,7 +27,9 @@
 
 #include <tuple>
 
+#include "base/command_line.h"
 #include "build/build_config.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "third_party/blink/public/common/thread_safe_browser_interface_broker_proxy.h"
 #include "third_party/blink/public/platform/browser_interface_broker_proxy.h"
 #include "third_party/blink/public/platform/platform.h"
@@ -96,6 +98,10 @@ void SpeechSynthesis::OnSetVoiceList(
 }
 
 const HeapVector<Member<SpeechSynthesisVoice>>& SpeechSynthesis::getVoices() {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          ::switches::kFingerprint)) {
+    return voice_list_;
+  }
   // Kick off initialization here to ensure voice list gets populated.
   std::ignore = TryEnsureMojomSynthesis();
   return voice_list_;
