--- /tmp/chromium_orig/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc	2026-05-27 04:39:02
+++ /Volumes/Lexar-1T/mk_ch/chromium/src/third_party/blink/renderer/modules/gamepad/navigator_gamepad.cc	2026-08-01 16:27:31
@@ -26,6 +26,8 @@
 #include "third_party/blink/renderer/modules/gamepad/navigator_gamepad.h"
 
 #include "base/auto_reset.h"
+#include "base/command_line.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "device/gamepad/public/cpp/gamepads.h"
 #include "third_party/blink/public/mojom/frame/user_activation_notification_type.mojom-blink.h"
 #include "third_party/blink/public/platform/task_type.h"
@@ -88,6 +90,10 @@
 HeapVector<Member<Gamepad>> NavigatorGamepad::getGamepads(
     Navigator& navigator,
     ExceptionState& exception_state) {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          ::switches::kFingerprint)) {
+    return HeapVector<Member<Gamepad>>();
+  }
   if (!navigator.DomWindow()) {
     // Using an existing NavigatorGamepad if one exists, but don't create one
     // for a detached window, as its subclasses depend on a non-null window.
