diff --git a/third_party/blink/renderer/core/loader/frame_fetch_context.cc b/third_party/blink/renderer/core/loader/frame_fetch_context.cc
--- a/third_party/blink/renderer/core/loader/frame_fetch_context.cc
+++ b/third_party/blink/renderer/core/loader/frame_fetch_context.cc
@@ -36,6 +36,7 @@
 
 #include "base/command_line.h"
 #include "base/feature_list.h"
+#include "components/ungoogled/ungoogled_switches.h"
 #include "base/memory/scoped_refptr.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/histogram_macros.h"
@@ -1004,6 +1005,11 @@
     const ResourceLoaderOptions& options) {
   AddClientHintsIfNecessary(resource_width, request);
   AddReducedAcceptLanguageIfNecessary(request);
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          ::switches::kFingerprint)) {
+    request.SetHttpHeaderField(http_names::kAcceptLanguage,
+                               AtomicString("en-US,en;q=0.9"));
+  }
 }
 
 bool FrameFetchContext::StartSpeculativeImageDecode(Resource* resource) {
